### DISCUSSION TITLE
Discontinuity in the phonon branches

### DISCUSSION THREAD

**Discussion by apandit2021** [ttadano/alamode#34] [Q&A] [answered]
Dear Dr. Tadano,

I am getting unusual fluctuations at/along some wave-vectors boundaries for several phonon branches as in the attached image below. I checked by changing some of the input parameters as well. But the pattern is still the same. Do you have any suggestions on this issue?

![SCPH-test](https://user-images.githubusercontent.com/84091914/126662416-c00400c3-bc88-4af2-b412-588cf8059c50.jpg)

Best,
Abhiyan

**Comment by ttadano** [comment]
Hi. Thank you for the report. Could you upload a higher resolution figure? I can't see the "fluctuation" you mentioned. Also, did you get a similar problem in the harmonic phonon calculation?

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

I think the image attached below should be clear enough. By the "fluctuation", I mean if you look some of the phonon branches closely at X- and Y k-points, mainly for the lower optical branches, there is sudden rise/fall at the intersection, which is not smooth. I didn't have this problem in the harmonic phonon calculations.
For your view, following is the input file I used for the scp calculation (I also checked by increasing the KMESH_*, but the problem is still there):
&general
  PREFIX = 551_scph
  MODE = SCPH
  FCSXML = *551_quartic_Opt.xml
  NKD = 2
  KD = * * 
  NONANALYTIC = 2
  BORNINFO = *.born
  TMIN = 0; TMAX = 700
  DT = 50
/

&scph
SELF_OFFDIAG = 0
MAXITER = 4000
MIXALPHA = 0.01
KMESH_INTERPOLATE = 5 5 1
KMESH_SCPH = 10 10 2
TOL_SCPH = 1.0e-8
/

&cell
50.00 # factor in Bohr unit
0.1753993041	0.0000000000	0.0000000000
0.0000000000	0.1730004475	0.0000000000
0.0000000000	0.0000000000	0.8353772789
/


&kpoint
  1  # KPMODE = 1: line mode
  G 0.0 0.0 0.0 X 0.5 0.0 0.0 51
  X 0.5 0.0 0.0 S 0.5 0.5 0.0 51
  S 0.5 0.5 0.0 Y 0.0 0.5 0.0 51
  Y 0.0 0.5 0.0 G 0.0 0.0 0.0 51
/


#&kpoint
# 2
# 30 30 2
#/


&analysis
 PRINTMSD = 1
/



![SCP-freq](https://user-images.githubusercontent.com/84091914/126679066-7ce0d9ae-b90b-4202-aa73-c21af0130b63.jpg)

**Comment by ttadano** [comment, marked as answer]
This is the expected behavior of the mixed-space approach ``NONANALYTIC=2``, which gives discontinuous dispersion curves at **_q_** points that are not commensurate with the supercell size. In your case, you used a 5x5x1 supercell, where the Brillouin zone boundaries such as X and Y are not commensurate. This problem should not happen when you used a 2m x 2m x 1 supercell (m: integer).

To solve this problem, please run an SCPH calculation with ``NONANALYTIC=2`` first and then switch to ``NONANALYTIC=3`` in the restart mode (``RESTART_SCPH=1``). The Ewald method (``NONANALYTIC=3``) is expected to give continuous curves. _Note_: This two-step procedure is necessary for the current implementation which does not support ``NONANALYTIC=3`` in the SCPH iteration.

BTW, I think setting ``KMESH_SCPH = 10 10 1`` gives the same result as ``KMESH_SCPH = 10 10 2`` with lower computational costs.

**Comment by apandit2021** [comment]
Dear Dr. Tadano,

Thank you for the explanation. Accordingly, I first performed the SCPH run and then used the restart mode calculation with the input tags as below. I already have PREFIX.scph_dymat generated, where I used ALAMODE-V-1.2.0. However, I am getting error with the input settings. Could you please look and let me know where I need the correction?

&general
  PREFIX = **
  MODE = RESTART
  FCSXML =551_quartic_Opt.xml
  NKD = 2
  KD = * *
  NONANALYTIC = 3
  BORNINFO = *.born
  TMIN = 0; TMAX = 700
  DT = 50
/

&scph
RESTART_SCPH = 1
SELF_OFFDIAG = 0
MAXITER = 4000
MIXALPHA = 0.01
KMESH_INTERPOLATE = 5 5 1
KMESH_SCPH = 10 10 1
TOL_SCPH = 1.0e-8
/

&cell
50.00 # factor in Bohr unit
0.1753993041	0.0000000000	0.0000000000
0.0000000000	0.1730004475	0.0000000000
0.0000000000	0.0000000000	0.8353772789
/


&kpoint
  1  # KPMODE = 1: line mode
  G 0.0 0.0 0.0 X 0.5 0.0 0.0 51
  X 0.5 0.0 0.0 S 0.5 0.5 0.0 51
  S 0.5 0.5 0.0 Y 0.0 0.5 0.0 51
  Y 0.0 0.5 0.0 G 0.0 0.0 0.0 51
/


#&kpoint
# 2
# 30 30 2
#/


&analysis
 PRINTMSD = 1
/


Best,
Abhiyan

**Reply by ttadano** [reply]
Please use ``MODE = SCPH``.

**Comment by apandit2021** [comment]
Dear Dr. Tadano,

I used MODE = SCPH also before. However, it gives the error as below:

..................
..................
Diagonalizing dynamical matrices for all k points ...
ERROR in add_longrange_matrix MESSAGE: Hermiticity of Dynamical matrix is broken.

Any help/suggestion is highly appreciated.

Best,
Abhiyan

**Reply by ttadano** [reply]
Please try the GitHub version (develop branch). I have recently updated the related source code.

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

Did you mean the ALAMODE-V-1.2.0? I tried with this version as well, but the same error is there as well.

Best,
Abhiyan

**Reply by ttadano** [reply]
I made some corrections after the release of 1.2.0, but they are still not packaged as a new release. For the moment, please ``git pull`` the changes in the develop branch and see if the issue is fixed.

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

The issue is resolved now. Thank you very much.

Best,
Abhiyan
